client/starwhale/api/_impl/job.py

```diff
@@ -187,6 +193,27 @@ def decorator(func: t.Callable) -> t.Callable:
 
                 key_name_needs.append(f"{n.__module__}:{n.__qualname__}")
 
+            ext_cmd_args = ""
+            parameters_sig = []
+            #  user defined handlers i.e. not predict/evaluate/fine_tune
+            if not build_in:
+                sig = inspect.signature(func)
+                parameters_sig = [
+                    {
+                        "name": p[0],
+                        "required": p[1].default is inspect._empty
+                        or (
+                            isinstance(p[1].default, HanderInput)
+                            and p[1].default.required
+                        ),
+                        "multiple": isinstance(p[1].default, ListInput),
+                    }
+                    for idx, p in enumerate(sig.parameters.items())
```

Comment:
`for idx, p in enumerate(sig.parameters.items())` -->
`for idx, (k, v) in enumerate(sig.parameters.items())`

k == p[0]
v == p[1]

Reply:
done